siocli/lib/SqlDump.cpp

```diff
@@ -0,0 +1,796 @@
+{
+    std::uint32_t clobLength = 0;
+    // Read length
+    if (!input.ReadVarint32(&clobLength))
+        throw std::runtime_error("Can't read string value length");
+
+    // Read sample
+    std::string str;
+    str.resize(clobLength);
+
+    if (!input.ReadRaw(str.data(), clobLength)) throw std::runtime_error("String data read failed");
+
+    return str;
+}
+
+std::int8_t readInt8(google::protobuf::io::CodedInputStream& input)
+{
+    std::int8_t data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read int8 failed");
+
+    return data;
+}
+
+std::int8_t readInt16(google::protobuf::io::CodedInputStream& input)
+{
+    std::int16_t data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read int16 failed");
+
+    return data;
+}
+
+std::int32_t readInt32(google::protobuf::io::CodedInputStream& input)
+{
+    std::int32_t data = 0;
+    if (!input.ReadVarint32(reinterpret_cast<std::uint32_t*>(&data)))
+        throw std::runtime_error("Read int32 failed");
+
+    return data;
+}
+
+std::int64_t readInt64(google::protobuf::io::CodedInputStream& input)
+{
+    std::int64_t data = 0;
+    if (!input.ReadVarint64(reinterpret_cast<std::uint64_t*>(&data)))
+        throw std::runtime_error("Read int64 failed");
+
+    return data;
+}
+
+std::uint8_t readUInt8(google::protobuf::io::CodedInputStream& input)
+{
+    std::int8_t data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read int8 failed");
+
+    return data;
+}
+
+std::uint8_t readUInt16(google::protobuf::io::CodedInputStream& input)
+{
+    std::uint16_t data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read int16 failed");
+
+    return data;
+}
+
+std::uint32_t readUInt32(google::protobuf::io::CodedInputStream& input)
+{
+    std::uint32_t data = 0;
+    if (!input.ReadVarint32(&data)) throw std::runtime_error("Read int32 failed");
+
+    return data;
+}
+
+std::uint64_t readUInt64(google::protobuf::io::CodedInputStream& input)
+{
+    std::uint64_t data = 0;
+    if (!input.ReadVarint64(&data)) throw std::runtime_error("Read int64 failed");
+
+    return data;
+}
+
+float readFloat(google::protobuf::io::CodedInputStream& input)
+{
+    float data = 0;
+    if (!input.ReadLittleEndian32(reinterpret_cast<std::uint32_t*>(&data)))
+        throw std::runtime_error("Read float failed");
+
+    return data;
+}
+
+double readDouble(google::protobuf::io::CodedInputStream& input)
+{
+    double data = 0;
+    if (!input.ReadLittleEndian64(reinterpret_cast<std::uint64_t*>(&data)))
+        throw std::runtime_error("Read double failed");
+
+    return data;
+}
+
+bool readBool(google::protobuf::io::CodedInputStream& input)
+{
+    std::uint8_t data = 0;
+    if (!input.ReadRaw(&data, sizeof(data))) throw std::runtime_error("Read bool failed");
+
+    return data > 0;
+}
+
+std::string readValue(google::protobuf::io::CodedInputStream& input, ColumnDataType columnDataType)
```

Comment:
This one should **not** go into `CustomCodedInputStream`, please leave it here as is.